### ISSUE TITLE
Runtime Error on Large File Upload - Double to Int Conversion Issue

### ISSUE THREAD

**Issue by SrinivasKalagotla** [aws-amplify/amplify-swift#3378]
### Describe the bug

When attempting to upload a large audio file (around 40 minutes in length) to AWS S3 using AWS Amplify from an Apple Watch-only OS application, the application crashes with a runtime error. The error occurs specifically in the Bytes enum of the AWS SDK, where there is an attempt to convert a Double value to an Int. This conversion fails when the Double value exceeds Int.max. The upload process is working for small audio files.

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Record an audio file of a significant length (approximately 40 minutes) on apple watch device.
2. Attempt to upload this file to AWS S3 using AWS Amplify.
3. Observe that for large files, the application crashes with a runtime error.
```

### Expected behavior

The expectation is that large files should be uploaded without causing a runtime error due to data type conversion limitations in the SDK.

### Amplify Framework Version

2.22.0

### Amplify Categories

Storage

### Dependency manager

Swift PM

### Swift version

Latest Version

### CLI version

Latest Version

### Xcode version

15.0.1

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>

Stopping recording. Current recording URL: Optional(file:///var/mobile/Containers/Data/Application/4AA4E254-97F6-4A79-88B8-D0F86B021C37/Documents/G_2023-11-23_20_12.m4a)
...
🔍 File exists, proceeding with upload.
File size to upload: 5907509 bytes
...
Swift/arm64_32-apple-watchos.swiftinterface:34421: Fatal error: Double value cannot be converted to Int because the result would be greater than Int.max


Error Location:

Bytes.swift - enum Bytes {
    case terabytes(Int)
    ...
    var bytes: Int {
        switch self {
        case .terabytes(let tb):
            return Int(pow(1_024.0, 4.0)) * tb
        ...
    }
}

Error Message
Thread 18: Fatal error: Double value cannot be converted to Int because the result would be greater than Int.max

</details>
```

### Is this a regression?

Yes

### Platforms

watchOS

### OS Version

Latest Version

### Device

Apple Watch 8

### Additional context

The issue seems to be with the handling of large numbers in the AWS Amplify SDK, particularly for large file uploads from an Apple Watch OS application. This issue was not encountered with smaller file sizes.

**Comment by ruisebas** [comment]
Hi @SrinivasKalagotla, thanks for opening this issue.

We will take a look and post any updates here.

**Comment by ruisebas** [comment]
Hi @SrinivasKalagotla, I've pushed a potential fix for this issue to the [**`fix/watchos_storage`**](https://github.com/aws-amplify/amplify-swift/tree/fix/watchos_storage) branch.

Would you be able to validate if using said branch in your app prevents the crashes you've been experiencing? 

Thanks!

**Comment by 5d** [comment]
Hi @SrinivasKalagotla ,

The Amplify CLI is designed for handling AWS resources within Amplify. For managing dependencies in your Xcode project, it is recommended to utilize Swift Package Manager.

**Comment by SrinivasKalagotla** [comment]
Thank you. Will test it and update

**Comment by SrinivasKalagotla** [comment]
Thank you, @ruisebas , for your diligent work on the app. I've successfully tested the voice recording for 45 minutes, and it's great to see everything smoothly transferring to the AWS S3 bucket.

Your fix to switch from int to UInt64 seems promising. I trust your expertise in this matter and look forward to seeing how this enhances the amplify's functionality. I appreciate your hard work and the effort you've put into this project.

**Comment by ruisebas** [comment]
Hi @SrinivasKalagotla, thanks for testing it on your end! I'm glad the issue's been resolved.
We'll aim to release a new Amplify version with the fix during next week.

Thanks!

**Comment by SrinivasKalagotla** [comment]
Hi @ruisebas , 

A small update. I have tested voice recording for 1hr 30 min duration and the file is not transferred to the S3 bucket.

I didn't connect my app to Xcode during this time so I am not sure what error did the app gave. I will debug the issue again and will provide you update on the error.

Thank you

**Comment by ruisebas** [comment]
Thanks for confirming! We'll close this issue once its released.

**Comment by ruisebas** [comment]
The fix has been released in [2.22.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.22.1).
Closing this issue.
